src/sentry/rules/conditions/tagged_event.py

```diff
@@ -64,71 +64,80 @@ def _passes(self, raw_tags: Sequence[tuple[str, Any]]) -> bool:
+            if k.lower() == option_key or tagstore.backend.get_standardized_key(k) == option_key
         )
 
-        if match == MatchType.EQUAL:
-            for t_value in values:
-                if t_value == value:
+        if option_match == MatchType.EQUAL:
+            for t_value in tag_values:
+                if t_value == option_value:
                     return True
             return False
 
-        elif match == MatchType.NOT_EQUAL:
-            for t_value in values:
-                if t_value == value:
+        elif option_match == MatchType.NOT_EQUAL:
+            for t_value in tag_values:
+                if t_value == option_value:
                     return False
             return True
 
-        elif match == MatchType.STARTS_WITH:
-            for t_value in values:
-                if t_value.startswith(value):
+        elif option_match == MatchType.STARTS_WITH:
+            for t_value in tag_values:
+                if t_value.startswith(option_value):
                     return True
             return False
 
-        elif match == MatchType.NOT_STARTS_WITH:
-            for t_value in values:
-                if t_value.startswith(value):
+        elif option_match == MatchType.NOT_STARTS_WITH:
+            for t_value in tag_values:
+                if t_value.startswith(option_value):
                     return False
             return True
 
-        elif match == MatchType.ENDS_WITH:
-            for t_value in values:
-                if t_value.endswith(value):
+        elif option_match == MatchType.ENDS_WITH:
+            for t_value in tag_values:
+                if t_value.endswith(option_value):
                     return True
             return False
 
-        elif match == MatchType.NOT_ENDS_WITH:
-            for t_value in values:
-                if t_value.endswith(value):
+        elif option_match == MatchType.NOT_ENDS_WITH:
+            for t_value in tag_values:
+                if t_value.endswith(option_value):
                     return False
             return True
 
-        elif match == MatchType.CONTAINS:
-            for t_value in values:
-                if value in t_value:
+        elif option_match == MatchType.CONTAINS:
+            for t_value in tag_values:
+                if option_value in t_value:
                     return True
             return False
 
-        elif match == MatchType.NOT_CONTAINS:
-            for t_value in values:
-                if value in t_value:
+        elif option_match == MatchType.NOT_CONTAINS:
+            for t_value in tag_values:
+                if option_value in t_value:
                     return False
             return True
 
+        elif option_match == MatchType.IS_IN:
+            values = option_value.replace(" ", "").split(",")
+            for t_value in tag_values:
+                if t_value in values:
+                    return True
+            return False
```

Comment:
Minor perf improvements to this logic:
```suggestion
            values = set(option_value.replace(" ", "").split(","))
            return any(t_value in values for t_value in tag_values)
```
This will ensure that if a tag is listed multiple times, we will only check it once. This also reduces the runtime from O(n * m) to O(n) because the `any` check will return true on the first match case vs the in check iterating `values`.

Reply:
Nice, i'll update this for all condition match types